src/GafferArnold/ArnoldShader.cpp

```diff
@@ -128,12 +128,37 @@ void ArnoldShader::loadShader( const std::string &shaderName, bool keepExistingV
 
 	const bool isLightShader = AiNodeEntryGetType( shader ) == AI_NODE_LIGHT;
 	namePlug()->setValue( AiNodeEntryGetName( shader ) );
-	typePlug()->setValue(
-		isLightShader ? "ai:light" : "ai:surface"
-	);
+
+	int aiOutputType = AI_TYPE_POINTER;
+	string type = "ai:light";
+	if( !isLightShader )
+	{
+		bool isLightFilter = false;
+		const CompoundData *metadata = ArnoldShader::metadata();
+		if( metadata )
+		{
+			const StringData *shaderTypeData = static_cast<const StringData*>( metadata->member<IECore::CompoundData>( "shader" )->member<IECore::Data>( "shaderType" ) );
+			if( shaderTypeData )
+			{
+				isLightFilter = shaderTypeData->readable() == "lightFilter";
```

Comment:
When I suggested the "shaderType" metadata my intention was that it be a more generic mechanism than the "isLightFilter" mechanism we started with, but this is undoing that. Can we not just say `type = "ai:" + shaderTypeData->readable()`?